wordpress-mobile/WordPress-iOS
# Posts & Pages: Pages context menu part 1

Part of #21873

## Description
- Adds a context menu to the pages cell
- The set parent, set as homepage, and set as post page actions will be handled in the next PR

## Notes
- For now, I've copied over the actions logic as to avoid introducing regressions. We've discussed the unreliability of ReachabilityUtils, but I'd prefer to address any enhancements in a future PR

<img src="https://github.com/wordpress-mobile/WordPress-iOS/assets/6711616/96644a74-22ac-4da7-ab54-facc0a370561" width=200>

## How to test
- Go to the pages list screen
- Tap on the ellipsis button
- ✅ Verify a context menu appears
- Tap on actions
- ✅ Verify the selected action work as expected

## Regression Notes
1. Potential unintended areas of impact: pages actions

2. What I did to test those areas of impact (or what existing automated tests I relied on): added tests for PagesMenuViewModel

3. What automated tests I added (or what prevented me from doing so): PagesMenuViewModel

PR submission checklist:


UI Changes testing checklist:
